lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -23,12 +28,36 @@ class TestResultsPush:
     URL = "http://pbench.example.com/api/v1"
 
     @staticmethod
-    def add_http_mock_response(code: HTTPStatus = HTTPStatus.OK):
-        responses.add(
-            responses.PUT,
-            f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
-            status=code,
-        )
+    def add_http_mock_response(
+        status_code: HTTPStatus = None, message: Union[str, Dict, type(None)] = None
+    ):
+        if status_code:
+            if message is None:
+                responses.add(
+                    responses.PUT,
+                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
+                    status=status_code,
+                )
+            elif isinstance(message, dict):
+                responses.add(
+                    responses.PUT,
+                    f"{TestResultsPush.URL}/upload/{os.path.basename(tarball)}",
+                    status=status_code,
+                    json=message,
+                )
+            # I can only add a `json' attribute to the response. Trying `text' or `reason' makes
```

Comment:
I think that `reason` is implicit in the HTTP status, not really a separate `Response` attribute. I'm not exactly sure what distinguishes `text` from `data`, but I kinda get the impression that text *is* just `data`, but decoded to unicode rather than as bytes. So I think `data="a random text string"` ought to appear as `response.text`...

Reply:
OK - I'm not going to worry about adding text/reason to these tests this time around. I'll leave it for later.

Reply:
As @webbnh suggested, `body` seems to do the trick, so I added some test cases.